src/utils.ts

```diff
@@ -19,3 +32,22 @@ export function isDir (path: string | URL): boolean {
     return false
   }
 }
+
+export function isFile (path: string | URL): boolean {
+  try {
+    const info = Deno.lstatSync(path)
+    return info.isFile
+  } catch (_e) {
+    return false
+  }
+}
+
+export function isHashKey (key: string): boolean {
+  return key.length === HASH_LENGTH && !key.startsWith('head=') && !key.startsWith('name=')
+}
```

Comment:
Some hashes might have longer than `HASH_LENGTH` hash lengths. So maybe we should give up on trying to identify a hash key, and rather check for known things that aren't hash keys. E.g.:

```js
function isNotHashKey(key: string): boolean {
  return key.startsWith('head=') || key.startsWith('name=')
}
```

Reply:
I'm not fan of `isNot-` names but if using `HASH_LENGTH` doesn't make sense then I agree